### PR TITLE
fix: switch non-token based css variables to less variables

### PR DESCRIPTION
**Pull request by anttio**
## 🖼 Context

Other than colour and spacing CSS properties are not fully defined and documented as design tokens. We want to stop exposing other custom properties and have a clear and defined strategy for using tokens in the future.

## 🚀 Changes

* Standardised imports of less from neptune-css into components to refer to src, not dist (less chance of the build breaking)
* Updated the variable structure in neptune-css:
   * variables we expose live in `variables` and are exported to `less`
   * custom props we expose live in `props` and are exported to `props`
* Removed the `neptune-legacy` bundle
* Added a new `legacy-custom-props.css` file to hold all the custom props we've removed from `neptune-css`
* Exposed a `neptune-tokens.css` file that holds all the tokens from `neptune-tokens`
* Documented how to use custom props correctly, along with the extra setup required to use custom-media
* Replaced all custom properties that are not using either colour, or spacing tokens from neptune-tokens with LESS variables

## ✅ Checklist

- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/master/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/css/src/less/flex.less [hunk 1]
```diff
@@ -73,7 +68,7 @@
   max-width: 50%;
 
   &--md {
-    @media all and (max-width: @screen-md) {
+    @media (--screen-md-max) {
```
Comment (nabaroa) [inline packages/css/src/less/flex.less:71]: This meant to be `--screen-sm-max` or `--screen-md`? 

- @screen-md: 768px;
- --screen-md (min-width: 768px);
- --screen-sm-max (max-width: 767px);

File: packages/css/src/less/flex.less [hunk 2]
```diff
@@ -85,7 +80,7 @@
   max-width: 100%;
 
   &--md {
-    @media all and (max-width: @screen-md) {
+    @media (--screen-md-max) {
```
Comment (nabaroa) [inline packages/css/src/less/flex.less:83]: Same as previous: This meant to be --screen-sm-max or --screen-md?

File: packages/css/src/less/flex.less (hunk 1 above)
Comment (anttio) [reply packages/css/src/less/flex.less:71]: This should be `max-width: 768px` and currently there's no `custom-media` defined for this. Let's use LESS variable.

Comment (nabaroa) [reply packages/css/src/less/flex.less]: Yes, `--screen-sm-max` works here.

File: packages/css/src/less/chip.less [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
-    background: @brand-smoke;
+    background: @color-base-smoke-mid;
   }
 
   .chip-remove {
     opacity: 0.6;
-    margin-left: (@spacer * 2);
-    height: (@spacer * 2);
+    margin-left: var(--size-16);
+    height: calc(var(--size-8) * 2);
```
Comment (anttio) [inline packages/css/src/less/chip.less:21]:
```suggestion
    height: var(--size-16);
```

File: packages/css/src/less/grid-layout.less [hunk 4]
```diff
@@ -198,11 +198,11 @@
 
 [class^='grid--'],
 [class*=' grid--'] {
-  grid-gap: calc(var(--spacer) * 4);
+  grid-gap: calc(var(--size-8) * 4);
```
Comment (anttio) [inline packages/css/src/less/grid-layout.less:201]:
```suggestion
  grid-gap: var(--size-32);
```

File: packages/css/src/less/grid-layout.less [hunk 5]
```diff
@@ -198,11 +198,11 @@
...
 
   [class^='grid--'],
   [class*=' grid--'] {
-    grid-gap: calc(var(--spacer) * 2);
+    grid-gap: calc(var(--size-8) * 2);
```
Comment (anttio) [inline packages/css/src/less/grid-layout.less:205]:
```suggestion
    grid-gap: var(--size-16);
```

File: packages/css/src/less/grid-layout.less [hunk 6]
```diff
@@ -223,7 +223,7 @@
 }
 
 .grid-row-gap--6 {
-  grid-row-gap: calc(var(--spacer) * 6);
+  grid-row-gap: calc(var(--size-8) * 6);
```
Comment (anttio) [inline packages/css/src/less/grid-layout.less:226]:
```suggestion
  grid-row-gap: var(--size-48);
```

File: packages/docs/pages/styles/tokens/Breakpoints.mdx [hunk 7]
```diff
@@ -1,4 +1,41 @@
...
+      importFrom: './node_modules/@transferwise/neptune-css/src/props/custom-media.css',
+    },
+  },
+};
+```
+
+## Usage
+
+```js
+// You only need this one in your app; props are added to :root and available everywhere
```
Comment (nabaroa) [inline packages/docs/pages/styles/tokens/Breakpoints.mdx:30]: Not sure if this explanation is totally accurate. The custom media are not inside the :root.

File: packages/docs/pages/styles/tokens/Colors.mdx [hunk 8]
```diff
@@ -1,3 +1,5 @@
+import { DocLink } from '../../../utils/pageUtils';
```
Comment (nabaroa) [inline packages/docs/pages/styles/tokens/Colors.mdx:1]: 76 line of this file. `Brand colors are available as `LESS` variables from `variables/_colors.less`. Not accurate anymore.

File: packages/css/src/less/circles.less [hunk 9]
```diff
@@ -2,7 +2,6 @@
 
 // variables
 @import (reference) 'bootstrap/less/variables.less';
```
Comment (nabaroa) [inline packages/css/src/less/circles.less:4]: legacy-variables.less?

File: packages/css/src/less/drawer.less [hunk 10]
```diff
@@ -24,8 +22,8 @@
   }
 
   .drawer-header {
-    height: calc(var(--spacer) * 7);
-    min-height: calc(var(--spacer) * 7);
+    height: calc(var(--size-8) * 7);
```
Comment (nabaroa) [inline packages/css/src/less/drawer.less:25]: --size-56?

File: packages/css/src/less/drawer.less [hunk 11]
```diff
@@ -56,8 +54,8 @@
 @media (--screen-md) {
   .drawer {
     .drawer-header {
-      height: calc(var(--spacer) * 8);
-      min-height: calc(var(--spacer) * 8);
+      height: calc(var(--size-8) * 8);
```
Comment (nabaroa) [inline packages/css/src/less/drawer.less:57]: --size-64?

**Review by nabaroa — COMMENTED** [review]
Can we add these in the legacy-custom-props.less? We are using them in some products already:
:root {
  --space-4: 4px;
  --space-8: 8px;
  --space-12: 12px;
  --space-16: 16px;
  --space-24: 24px;
  --space-32: 32px;
  --space-40: 40px;
  --space-48: 48px;
  --space-56: 56px;
  --space-64: 64px;
}

File: packages/css/src/less/circles.less (hunk 9 above)
Comment (anttio) [reply packages/css/src/less/circles.less:4]: There's still bunch of components importing variables from Bootstrap. These are components that would need more refactoring and we can revisit the imports during that. As this PR is quite big already I would leave these for later.

File: packages/docs/pages/styles/tokens/Breakpoints.mdx (hunk 7 above)
Comment (helendurranttw) [reply packages/docs/pages/styles/tokens/Breakpoints.mdx:30]: Ahh good point. I'll update.

**Review by helendurranttw — APPROVED** [review]

**Pull request merged**